### PR TITLE
feat: Log key prefix controls for Trainer, WandbLogger, and CometLogger 

### LINKED ISSUE #21782: Log key prefix controls for Trainer, WandbLogger, and CometLogger
### Description & Motivation

Lightning automatically injects metric keys like `epoch` (via `Trainer`) and `global_step` (via
`WandbLogger`) into the logged metrics stream. These keys are currently hardcoded and cannot be
prefixed or renamed without monkey-patching internals.

This creates a friction point when users want a clean, consistent metric namespace. For example,
when using W&B or Comet, it is common to organise metrics under prefixes like `trainer/`,
`train/`, `val/`, etc. User-logged metrics can already be namespaced freely via `self.log(...)`,
but Lightning's own injected keys (`epoch`, `trainer/global_step`) sit outside that control. 
They are either hardcoded into a different namespace or into no namespace at all.

A concrete pain point: `WandbLogger` hardcodes `trainer/global_step` as the step axis key, but
`Trainer` logs the epoch metric as bare `epoch`. There is no way to align these without modifying
Lightning internals. This means manual work on each new experiment to align the dashboards in 
W&B.

### Pitch

Add controls so that each component that *generates* metric keys exposes a parameter to configure 
those keys without touching any metric keys the user provides.

The design principle:

> `log_key_prefix` prefixes the keys *generated by that component*.

### Proposed API

**`Trainer(log_key_prefix=...)`**

Prefixes Trainer-generated metric keys. Currently the only such key is the automatic `epoch`
metric injected at each logging step.

```python
Trainer(log_key_prefix="trainer/")
# logs: "trainer/epoch"

Trainer(log_key_prefix="")    # produces bare "epoch", the previous behaviour, backwards compatible.
Trainer(log_key_prefix=None)  # produces bare "epoch", the previous behaviour, backwards compatible.
Trainer()                     # produces bare "epoch", the previous behaviour, backwards compatible.
```

**`WandbLogger(log_key_prefix=...)`**

Controls the W&B-generated step key used in `define_metric` and `experiment.log`. The existing
`prefix=` argument is unchanged and continues to apply to the pass-through metric keys.

```python
WandbLogger(log_key_prefix="trainer/") # Produces "trainer/global_step", the previous behaviour.

WandbLogger(log_key_prefix="")    # Produces bare "global_step"
WandbLogger(log_key_prefix=None)  # Produces bare "global_step"
WandbLogger()                     # Produces bare "global_step", NOT backwards compatible.
```

**`CometLogger(epoch_key=...)`**

Controls which incoming metric key is extracted and forwarded to Comet's dedicated `epoch=`
argument. `None` disables epoch extraction entirely.

```python
CometLogger() # Matches Trainer(), extracting "epoch", so it's backward compatible.
CometLogger(epoch_key="epoch") # Extracts "epoch".
CometLogger(epoch_key="trainer/epoch") # Extracts "trainer/epoch", matching Trainer(log_key_prefix="trainer/")
CometLogger(epoch_key=None) # Disables special epoch extraction.
```

### Aligning all Lightning-generated keys under one namespace

```python
Trainer(log_key_prefix="trainer/")
WandbLogger(log_key_prefix="trainer/")
CometLogger(epoch_key="trainer/epoch")
```

This places `trainer/epoch` and `trainer/global_step` in the same namespace while leaving all
user metrics untouched.

### Alternatives

I explored having a dictionary of namespaces in Trainer that other modules could pick from so
they would be automatically aligned, but it was creating a lot of coupling. WandbLogger needed
a backreference to the trainer to pick up the log_key_prefix for example.

If this is an approach that interests you, I can explore it. 

### Additional context

_No response_

cc @lantiga

### PR DESCRIPTION
**Pull request by pupeno**
## What does this PR do?

Fixes #21782

_This PR has been manually self reviewed_

Adds three independent parameters that let users control the metric keys generated by Lightning's
own components, without touching user-provided metric names:

- `Trainer(log_key_prefix=...)`: prefixes Trainer-generated metric keys. Currently the only such
  key is the automatic `epoch` metric injected into every logging step. Default `None`/`""`
  preserves previous behaviour.

- `WandbLogger(log_key_prefix=...)`: controls the W&B-generated step key used in `define_metric`
  and `experiment.log`. Previously hardcoded to `"trainer/global_step"`; now configurable,
  defaulting to `"global_step"`. The existing `prefix=` argument is unchanged and continues to
  apply only to pass-through metric keys.

- `CometLogger(epoch_key=...)`: controls which incoming metric key is extracted and forwarded to
  Comet's dedicated `epoch=` argument. Previously hardcoded to `"epoch"`. `None` disables epoch
  extraction entirely. Default `"epoch"` preserves previous behaviour.

The design principle: each `log_key_prefix` (or `epoch_key`) applies only to keys *generated by
that component*, never to keys the user logged via `self.log(...)` or `logger.log_metrics(...)`.

This makes it straightforward to align all Lightning-generated keys under a shared namespace:

```python
Trainer(log_key_prefix="trainer/")
WandbLogger(log_key_prefix="trainer/")
CometLogger(epoch_key="trainer/epoch")
```

**Breaking change:** `WandbLogger` previously hardcoded `"trainer/global_step"` as the step axis
key. With this PR, the default is bare `"global_step"`. Pass `log_key_prefix="trainer/"` to
restore the previous behaviour.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Comment by deependujha** [comment]
Thanks for the thorough proposal and PR, the namespace inconsistency you're describing is real, and the Trainer-side piece is clean.

Would you be open to splitting this into two PRs? I think it'll move faster:

**PR 1: Trainer only.** The `log_key_prefix` kwarg on `Trainer` plus the one-line change in `logger_connector.py` (`setdefault(f"{prefix}epoch", ...)`). This part is fully additive and backward-compatible — defaults to bare `epoch`, so nothing changes for existing users. And because the key is injected upstream in the logger connector before any logger sees it, this benefits every logger downstream without touching any logger file. That makes it reviewable on the Trainer side independently.

**PR 2: per-logger keys.** Each logger's handling of the keys it injects itself. Those touch logger files and need the respective codeowners, so keeping them separate avoids blocking the Trainer change on that review.

If you split it, the Trainer PR should be straightforward to land. Thanks again.

**Comment by pupeno** [comment]
Ah, yes, that makes sense. I'll have it in a moment. Do you want me to create separate github issues or only separate PRs?

**Comment by pupeno** [comment]
Here are the PRs. I can also create the github issues if you want:

* https://github.com/Lightning-AI/pytorch-lightning/pull/21784
* https://github.com/Lightning-AI/pytorch-lightning/pull/21785
* https://github.com/Lightning-AI/pytorch-lightning/pull/21786

**Comment by pupeno** [comment]
@deependujha any advice on how to get these PRs merged?
